test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -324,6 +414,151 @@ private void checkTransactionReadAndWrite() throws TxnFailedException {
+                    return null;
+                }).join();
+        
+        // Inserting 9 more key into the tables
+        for (int i = 1; i < TEST_MAX_KEYS; i++) {
+            val insert = new Insert(toKey(userName + i, USERNAME_SERIALIZER), SERIALIZER.serialize(testData + i));
+            testKVTables.update(insert).join();
+        }
+        // Validating the content of the KV table
+        for (int i = 0; i < TEST_MAX_KEYS; i++) {
+            assertTrue(testKVTables.exists(toKey(userName + i, USERNAME_SERIALIZER)).join());
+            assertEquals(SERIALIZER.deserialize(testKVTables.get(toKey(userName + i, USERNAME_SERIALIZER)).join().getValue()), testData + i);
+        }
+        // Listing all the entries of kv tables
+        val count = new AtomicInteger(0);
+        testKVTables.iterator()
+                .maxIterationSize(20)
+                .all()
+                .keys()
+                .forEachRemaining(tableKey -> {
+                    for (val user : tableKey.getItems()) {
+                        val key = USERNAME_SERIALIZER.deserialize(user.getPrimaryKey());
+                        log.info("User: {} ", key);
+                        count.incrementAndGet();
+                    }
+                }, this.executor).join();
+        assertEquals(TEST_MAX_KEYS, count.get());
+        // Updating the key to some new data and validating it
+        val put = new Put(toKey(userName + "0", USERNAME_SERIALIZER), SERIALIZER.serialize(testData));
+        testKVTables.update(put);
+        assertTrue(testKVTables.exists(toKey(userName + "0", USERNAME_SERIALIZER)).join());
+        assertEquals(SERIALIZER.deserialize(testKVTables.get(toKey(userName + "0", USERNAME_SERIALIZER)).join().getValue()), testData);
+
+        for (int i = 0; i < TEST_MAX_KEYS; i++) {
+            val delete = new Remove(toKey(userName + i, USERNAME_SERIALIZER));
+            testKVTables.update(delete).join();
+            assertFalse(testKVTables.exists(toKey(userName + i, USERNAME_SERIALIZER)).join());
+        }
+        // Deleting key value table and validating it.
+        assertTrue(keyValueTableManager.deleteKeyValueTable(scopeName, keyValueTableName));
+    }
+
+    /**
+     * This method writes and reads large events (8MB) and validates them.
+     */
+    @Test(timeout = 20000)
+    private void checkWriteAndReadLargeEvent() {
+        String scopeName = "largeevent-test-scope";
+        String streamName = "largeevent-test-stream";
+        createScopeAndStream(scopeName, streamName);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<ByteBuffer> writer = clientFactory.createEventWriter(streamName, new ByteBufferSerializer(), EventWriterConfig.builder().enableLargeEvents(true).build());
+
+        byte[] payload = new byte[Serializer.MAX_EVENT_SIZE * 8];
```

Comment:
If you look at the actual value of `Serializer.MAX_EVENT_SIZE`:
https://github.com/pravega/pravega/blob/40067eed0438504be9c1ba86c7127ac762c26b29/client/src/main/java/io/pravega/client/stream/Serializer.java#L33

It seems to be 8MB already, so I think that just using `Serializer.MAX_EVENT_SIZE * 2` is enough to exercise the large event workflow.

Reply:
ok, yeah